File: app/schemas/branding.py
from pydantic import BaseModel

from app.services.branding_settings_service import (
    DEFAULT_ICON_URL,
    DEFAULT_LOGIN_SUBTITLE,
    DEFAULT_PRODUCT_NAME,
    DEFAULT_AGENT_NAME,
)


class BrandingSettings(BaseModel):
    enabled: bool = False
    product_name: str = DEFAULT_PRODUCT_NAME
    login_subtitle: str = DEFAULT_LOGIN_SUBTITLE
    icon_url: str = DEFAULT_ICON_URL
    hide_login_sso: bool = False
    hide_version_link: bool = False
    contact_markdown: str = ""
    copyright_text: str = ""
    default_agent_name: str = DEFAULT_AGENT_NAME


class BrandingSettingsUpdate(BrandingSettings):
    pass


class PublicBrandingResponse(BaseModel):
    enabled: bool = False
    product_name: str
    login_subtitle: str
    icon_url: str
    hide_login_sso: bool = False
    hide_version_link: bool = False
    contact_markdown: str = ""
    copyright_text: str = ""
    default_agent_name: str

File: frontend/src/constants/branding.ts
export const DEFAULT_PRODUCT_NAME = 'Hose·智能体平台'
export const DEFAULT_LOGIN_SUBTITLE = 'Hose Intelligent Agent Platform'
export const DEFAULT_ICON_URL = '/branding/icon.png'
export const DEFAULT_REPO_URL = 'https://github.com/RandyChen1985/nanzi-ai-agent-platform'
export const DEFAULT_AGENT_NAME = 'Hose · AI'

export interface PublicBranding {
  enabled: boolean
  product_name: string
  login_subtitle: string
  icon_url: string
  hide_login_sso: boolean
  hide_version_link: boolean
  contact_markdown: string
  copyright_text: string
  default_agent_name: string
}

export const DEFAULT_BRANDING: PublicBranding = {
  enabled: false,
  product_name: DEFAULT_PRODUCT_NAME,
  login_subtitle: DEFAULT_LOGIN_SUBTITLE,
  icon_url: DEFAULT_ICON_URL,
  hide_login_sso: false,
  hide_version_link: false,
  contact_markdown: '',
  copyright_text: '',
  default_agent_name: DEFAULT_AGENT_NAME,
}
